docs/docs/configuration/gstreamer.md

```diff
@@ -0,0 +1,324 @@
+
+**Note:** There are no advantages of GStreamer versus FFmpeg on non-Jetson devices.
+
+## Minimal GStreamer Configuration
+
+```yaml
+camera_name:
+  gstreamer:
+    inputs:
+      - path: rtsp://10.0.0.1:554/stream0
+        roles:
+          - detect
+  detect:
+    width: 1920
+    height: 1080
+```
+
+With the minimal configuration GStreamer integration will do the following:
+- Run `gst-inspect-1.0` to get the available plugins
+- Run `gst-discoverer-1.0` against the RTSP stream. The discovery process gets the audio and video stream codecs
+- Build the GStreamer pipeline based on the available plugins.
+- GStreamer automatically enable audio stream for recording if audio is available
+
+The default configuration should be enough for most of the cases. However, if you have multiple cameras, running `gst-discoverer-1.0` for each one might be pretty time-consuming. To avoid running `gst-discoverer-1.0`, you have to specify the video codec as well as the audio codec for the `record` role.
+
+
+## GStreamer configuration with video and audio codecs set
+
+```yaml
+camera_name:
+  gstreamer:
+    video_format: video/x-h265
+    audio_format: audio/x-alaw
+    inputs:
+      - path: rtsp://10.0.0.1:554/stream0
+        # video_format: video/x-h265
+        # audio_format: audio/x-alaw
+        roles:
+          - record
+
+      - path: rtsp://10.0.0.1:554/stream1
+        # video_format: video/x-h265
+        roles:
+          - detect
+  detect:
+    width: 1920
+    height: 1080
+```
+
+This setup is much more reliable, as it won't trigger the call to the `gst-discoverer-1.0` which might timeout occasionally. If you have different audio or video formats for different streams, you can override the defaults for each input.
+
+### Supported Video Formats
+
+GStreamer integration does not limit you which video format to use. It solely depends on the GStreamer plugins. GStreamer integration is building the `parse` plugin by doing the following steps:
+
+- lowercase the video format
+- strip the optional `video/x-` prefix
+- add the remainder to the `parse` for the `recorder`
+- create"rtp{video_format}depay" element for the `decoder`
+
+That way, if you specified `video/x-h264` as a video format, GStreamer should support the `h264parse` and `rtph264depay` pipeline elements.
+
+### Supported Audio Formats
+
+As of now, only `audio/x-alaw` and `audio/mpeg` are supported. Audio formats require different pipelines.
+To add a new audio format, one has to update `AUDIO_PIPELINES` in the `gstreamer.py`.
+
+Alternatively, `audio_pipeline` element of either input or camera level can be added to specify a custom audio pipeline. 
+`audio_pipeline` has a priority over `audio_format`, e.g. if you set both `audio_format` and `audio_pipeline`, the `audio_pipeline` will be used for decoding audio.
+
+Audio settings make sense only for the `record` role of your camera.
+
+
+```yaml
+camera_name:
+  gstreamer:
+    video_format: video/x-h265
```

Comment:
Attempting to set `video_format` throws an error for me:

```
frigate    | *************************************************************
frigate    | *************************************************************
frigate    | ***    Your config file is not valid!                     ***
frigate    | ***    Please check the docs at                           ***
frigate    | ***    https://docs.frigate.video/configuration/index     ***
frigate    | *************************************************************
frigate    | *************************************************************
frigate    | ***    Config Validation Errors                           ***
frigate    | *************************************************************
frigate    | 1 validation error for FrigateConfig
frigate    | cameras -> back -> gstreamer -> video_format
frigate    |   extra fields not permitted (type=value_error.extra)
frigate    | Traceback (most recent call last):
frigate    |   File "/opt/frigate/frigate/app.py", line 302, in start
frigate    |     self.init_config()
frigate    |   File "/opt/frigate/frigate/app.py", line 76, in init_config
frigate    |     user_config = FrigateConfig.parse_file(config_file)
frigate    |   File "/opt/frigate/frigate/config.py", line 991, in parse_file
frigate    |     return cls.parse_obj(config)
frigate    |   File "/usr/local/lib/python3.8/dist-packages/pydantic/main.py", line 511, in parse_obj
frigate    |     return cls(**obj)
frigate    |   File "/usr/local/lib/python3.8/dist-packages/pydantic/main.py", line 331, in __init__
frigate    |     raise validation_error
frigate    | pydantic.error_wrappers.ValidationError: 1 validation error for FrigateConfig
frigate    | cameras -> back -> gstreamer -> video_format
frigate    |   extra fields not permitted (type=value_error.extra)
```

A search for it in your fork yields [zero results](https://github.com/yury-sannikov/frigate/search?q=video_format).